itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/connection/v3/SetOCForNewConnectionTest.java

```diff
@@ -0,0 +1,240 @@
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.reddeer.common.wait.WaitWhile;
+import org.eclipse.reddeer.eclipse.ui.views.properties.PropertySheet;
+import org.eclipse.reddeer.junit.execution.annotation.RunIf;
+import org.eclipse.reddeer.junit.runner.RedDeerSuite;
+import org.eclipse.reddeer.requirements.openperspective.OpenPerspectiveRequirement.OpenPerspective;
+import org.eclipse.reddeer.swt.impl.button.CancelButton;
+import org.eclipse.reddeer.swt.impl.button.FinishButton;
+import org.eclipse.reddeer.swt.impl.menu.ContextMenuItem;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
+import org.jboss.tools.common.reddeer.perspectives.JBossPerspective;
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistrySingleton;
+import org.jboss.tools.openshift.reddeer.enums.AuthenticationMethod;
+import org.jboss.tools.openshift.reddeer.requirement.CleanOpenShiftExplorerRequirement.CleanOpenShiftExplorer;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftCommandLineToolsRequirement;
+import org.jboss.tools.openshift.reddeer.utils.DatastoreOS3;
+import org.jboss.tools.openshift.reddeer.utils.OpenShiftLabel;
+import org.jboss.tools.openshift.reddeer.view.OpenShiftExplorerView;
+import org.jboss.tools.openshift.reddeer.view.resources.OpenShift3Connection;
+import org.jboss.tools.openshift.reddeer.wizard.v3.BasicAuthenticationSection;
+import org.jboss.tools.openshift.reddeer.wizard.v3.OAuthauthenticationSection;
+import org.jboss.tools.openshift.reddeer.wizard.v3.OpenShift3ConnectionWizard;
+import org.jboss.tools.openshift.ui.bot.test.application.v3.basic.AbstractTest;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Implements JBIDE-24933
+ * @author jkopriva@redhat.com
+ */
+@RunWith(RedDeerSuite.class)
+@OpenPerspective(JBossPerspective.class)
+//@OCBinary(cleanup = true)
+@CleanOpenShiftExplorer
+public class SetOCForNewConnectionTest extends AbstractTest {
+
+	private static final String OC_370_URL_LINUX = "https://github.com/openshift/origin/releases/download/v3.7.0/openshift-origin-client-tools-v3.7.0-7ed6862-linux-64bit.tar.gz";
+	private static final String OC_370_URL_WINDOWS = "https://github.com/openshift/origin/releases/download/v3.7.0/openshift-origin-client-tools-v3.7.0-7ed6862-windows.zip";
+	private static final String OC_370_URL_MAC = "https://github.com/openshift/origin/releases/download/v3.7.0/openshift-origin-client-tools-v3.7.0-7ed6862-mac.zip";
+	private static final String PROPERTY_OC_LOCATION = "OC client path";
+	private String pathToOC;
+	Path ocTestDirectory = Paths.get("target", "octest");
+
+	//@InjectRequirement
+	//private OpenShiftCommandLineToolsRequirement ocRequirement;
+
+	public String downloadOCBinary(boolean latestOC, String destinationDirectory) {
```

Comment:
I was kinda aiming for a suggestion that if we wont use that requirement directly, we might want to extract useful methods from it into some utility class and keep ocbinary requirement simple and just calling methods from such class. wdyt?

Reply:
@odockal we can move it, but I really do not see any use case for it (at least for now).

Reply:
Sure thing.. ;)